File: euclidean_primer/euclidean_primer__mod_calc.py
# 実行時間 20ms
n = int(input())
r = input().split()
a = int(r[0])
b = int(r[2])

if r[1] == "+":
    print((a % n + b % n) % n)
elif r[1] == "-":
    print((a % n - b % n + n) % n)
elif r[1] == "*":
    print((a % n) * (b % n) % n)
elif r[1] == "^":
    o = 1

    for _ in [0] * b:
        o *= a
        o %= n
    print(o)

File: euclidean_primer/euclidean_primer__mod_calc.cs
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        string[] r = ReadLine().Split( ' ' );
        int a = int.Parse( r[0] );
        int b = int.Parse( r[2] );

        switch( r[1] ) {
        case "+":
            WriteLine( ( a % n + b % n ) % n );
            break;
        case "-":
            WriteLine( ( a % n - b % n + n ) % n );
            break;
        case "*":
            WriteLine( ( a % n ) * ( b % n ) % n );
            break;
        case "^":
            ulong o = 1;

            foreach( int _ in Range( 0, b ) ) {
                o *= (ulong)a;
                o %= (ulong)n;
            }
            WriteLine( o );
            break;
        }
        return;
    }
}
